### PR TITLE
[TFLite] Added op tests for conv_activations

### PR DESCRIPTION
**Pull request by wwwind**
Op tests for conv operator with activation for 16x8 quantization mode.

Command to run them:
bazel run //tensorflow/lite/testing:generate_examples -- /tmp --zip_to_output=conv_activation --toco=$(pwd)/bazel-bin/tensorflow/lite/toco/toco

Follow up of #39543

### CODE REVIEW

**Review by suharshs — DISMISSED** [review]

**Comment by rthadur** [comment]
@wwwind can you please check below error ?
```
tensorflow/lite/testing/op_tests/conv_activation.py", line 143, in make_conv_relu_tests
    return make_conv_activation_tests(tf.nn.relu)(options)
  tensorflow/lite/experimental/mlir/testing/generate_examples.runfiles/google3/third_party/tensorflow/lite/testing/op_tests/conv_activation.py", line 129, in f
    expected_tf_failures=60)
  tensorflow/lite/experimental/mlir/testing/generate_examples.runfiles/google3/third_party/tensorflow/lite/testing/zip_test_utils.py", line 310, in make_zip_of_tests
    (zip_path, all_parameter_count, _MAX_TESTS_PER_ZIP))
RuntimeError: Too many parameter combinations for generating 'blaze-out/k8-opt/genfiles/third_party/tensorflow/lite/experimental/mlir/testing/conv_relu.zip'.
```

**Comment by wwwind** [comment]
Hi @rthadur Could you please mention which bazel test task is failing ? 
What bazel test command should I call to reproduce these failures ?
Thanks!

**Comment by rthadur** [comment]
@wwwind this tests are failing `https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/testing/zip_test_utils.py#L259` 

cc @suharshs

**Comment by wwwind** [comment]
I found the source of problems. They should be fixed now.
I used the following commands to run these tests locally:

bazel test --compilation_mode=opt //tensorflow/lite/testing:zip_test_conv_relu
bazel test --compilation_mode=opt //tensorflow/lite/testing:zip_test_conv_relu1
bazel test --compilation_mode=opt //tensorflow/lite/testing:zip_test_conv_relu6

If I combine parameters like it was before, then I hit the error:
"There are at least 744 combinations while the upper limit is 500.
Having too many combinations will slow down the tests."

Hi @suharshs, @rthadur Could you please re-approve this PR ? Thanks

**Review by rthadur — APPROVED** [review]

File: tensorflow/lite/testing/op_tests/conv_activation.py [hunk 1]
```diff
@@ -55,6 +56,20 @@ def f(options):
...
+            "quant_16x8": [False],
+            "dynamic_range_quantize": [False],
+        },
+        {
+            "input_shape": [[4, 6, 6, 1]],
+            "filter_shape": [[1, 1], [2, 3]],
+            "strides": [[1, 1, 1, 1], [1, 2, 3, 1]],
+            "dilations": [[1, 1, 1, 1], [1, 3, 2, 1]],
+            "padding": ["SAME", "VALID"],
+            "data_format": ["NHWC"],  # TODO(aselle): NCHW  would be good
```
Comment (rthadur) [inline tensorflow/lite/testing/op_tests/conv_activation.py:68]: can you please remove TODO

**Review by rthadur — CHANGES_REQUESTED** [review]

**Comment by wwwind** [comment]
Hi @rthadur ! I have removed TODO comment. Could you please re-approve ? Thanks!

**Review by rthadur — APPROVED** [review]

**Comment by jdduke** [comment]
Were you able to run this test locally using bazel? We're seeing a failure in the archive extraction:

```
F0923 04:56:40.382512    9627 generated_examples_zip_test.cc:244] Non-OK-status: archive_environment()->UnArchive(zip_file, tar_file, &decompress_tmp_dir) status: Unknown: unzip failed. stdout:
```

**Comment by jdduke** [comment]
I think you can repro by trying:

```
bazel test //tensorflow/lite/testing:zip_test_conv_relu_forward-compat
```

**Comment by gbaned** [comment]
@wwwind  Can you please check @jdduke's comments and keep us posted ? Thanks!

**Comment by wwwind** [comment]
Sure, I will check these failures!

**Comment by wwwind** [comment]
Hi @jdduke I can reproduce this error and it is due to the OS Error "File name too long", that happened during Archive -> Unzip.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/testing/generated_examples_zip_test.cc#L280

It has happened after recent updating with the master branch, when the new parameter 'dynamic_range_quantize=False' is added. 
with 'quant_16x8' is just too much.
Example of the file name:

conv_relu_forward-compat_channel_multiplier=2,constant_filter=True,data_format='NHWC',dilations=[1,1,1,1],dynamic_range_quantize=True,filter_shape=[3,3],fully_quantize=False,input_shape=[1,3,4,3],padding='VALID',quant_16x8=False,strides=[1,2,3,1]_tests.txt
        
There are two ways to solve:
- replace the name of some params with the shorter version - I don't know which one should I dio
- split arrays of dilations, input_shapes, strides into 1 per test/file
- implement a special check and a fallback code in the generated_examples_zip_test.cc

Could you please advise how should I go forward with this PR ? Thanks

**Comment by gbaned** [comment]
@jdduke Can you please assist on above comments from @wwwind. Thanks!

**Comment by gbaned** [comment]
@jdduke Any update on this CL? Please. Thanks!

**Comment by jdduke** [comment]
Apologies, this one had been assigned to @suharshs, will reassign for landing!

**Comment by gbaned** [comment]
Seems auto-merge is not happening but the changes are merged into master now, so we can close this. Thank you for the PR.

**Pull request closed without merging**